rcl/test/rcl/test_timer.cpp

```diff
@@ -304,14 +304,20 @@ TEST_F(TestTimerFixture, test_two_timers_ready_before_timeout) {
     ret = rcl_wait_set_fini(&wait_set);
     EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
   });
-  ret = rcl_wait(&wait_set, RCL_MS_TO_NS(100));
-  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
-  uint8_t nonnull_timers = 0;
-  for (uint8_t i = 0; i < wait_set.size_of_timers; i++) {
-    if (wait_set.timers[i] != NULL) {
-      nonnull_timers++;
+  auto start = std::chrono::system_clock::now();
+  uint8_t nonnull_timers = 0u;
+  do {
+    ret = rcl_wait(&wait_set, RCL_MS_TO_NS(100));
+    EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    for (uint8_t i = 0; i < wait_set.size_of_timers; i++) {
+      if (wait_set.timers[i] != NULL) {
+        nonnull_timers++;
+      }
     }
-  }
+  } while (
+    nonnull_timers == 0u ||
+    std::chrono::duration_cast<std::chrono::milliseconds>(
+      std::chrono::system_clock::now() - start).count() > 100u);
```

Comment:
@ivanpauno consider leaving a note explaining why this is necessary.

Reply:
Done https://github.com/ros2/rcl/pull/911/commits/257a9154069341cb236ee22e4a01d99e38c7a66f